concurrency/_index.md

```diff
@@ -1,7 +1,158 @@
+seo_title: "Jakarta Concurrency Specification | Jakarta EE"
 #<!--.................0123456789.123456789.123456789.123456789.123456789.123456789-->
 summary_sixty_char: "API for app level concurrency that integrates with container"
 project_id: "ee4j.cu"
 ---
+
+{{< grid/div class="specifications-content" >}}
+
+## Jakarta Concurrency Explained
+
+Application performance can make or break a user’s experience. Applications that provide a consistent experience and perform well are more likely to benefit user productivity.
+The Jakarta Concurrency specification targets application performance and usability by providing a standard API for developing concurrent application processes without compromising container integrity.
+It is not recommended to utilize traditional Java SE threads or timers within a Jakarta EE container because use of threads or timers may cause reliability issues and unexpected results.
+Jakarta Concurrency provides an API that does not intrude on container integrity, but rather, uses services that are managed by the container and matches the functionality provided by the `Java SE Concurrency Utilities`.
+
+Developers who are familiar with the `Java SE Concurrency Utilities` API will be able to leverage their existing knowledge to use this API, as there are many similarities.
+Containers extend the `java.util.concurrent` API by providing managed versions of `java.util.concurrent.ExecutorService` interfaces.
+All Jakarta EE compliant containers include default executor services, context services, and thread factories for performing particular tasks such as spawning asynchronous tasks and scheduling tasks to be executed at a specified time.
+However, it is also possible to develop custom services, if needed, and register them with a container.
+
+### Configuring Jakarta Concurrency
+
+Jakarta Concurrency is part of all Jakarta EE compatible containers. However, to explicitly add the Jakarta Concurrency API to a maven application, add the following dependency:
+
+```xml
+<dependency>
+    <groupId>jakarta.enterprise.concurrent</groupId>
+    <artifactId>jakarta.enterprise.concurrent-api</artifactId>
+    <version>3.0.2</version>
+</dependency>
+```
+
+### Asynchronous Tasks
+
+Jakarta EE compliant containers must contain a `ManagedExecutorService`, which extends the Java SE `ExecutorService`. Tasks can be passed to a `ManagedExecutorService`, and those tasks can then be performed within a container managed thread and passed back to the caller once complete.
+This process can occur asynchronously, allowing the application to continue performing other tasks while the `ManagedExecutorService` processes the task which had been handed off.
+The `ManagedExecutorService` returns a `Future` object to the caller, which will return a `boolean` indicating when the task has been completed.
+The caller can poll this `Future` object and then perform some action using the resulting object once the task has completed.
```

Comment:
```suggestion
The caller can poll this `Future` object and then perform some action using the resulting object once the task has completed. With a `CompletableFuture` or `CompletionStage`, the caller can chain dependent actions to the completion of the task rather than polling.
```